### DISCUSSION TITLE
Can I do mask validation?

### DISCUSSION THREAD

**Discussion by Amstrong** [react-hook-form/react-hook-form#5396] [Q&A] [answered]
**Well I have two questions almost related to the mask**

1. I would like to know if I can validate mask such as a date entered manually by the user. That it does not allow you to write more characters than those necessary for DD-MM-YYYY and that it allows me to validate these specific values, by ranges?

2. While the user is typing in a field, can I show him the format? that is: 23 - _ _ - _ _ _ _ referring to DD-MM-YYY

Thanks!

**Comment by bluebill1049** [comment, marked as answer]
Hey @Amstrong I would use an external mask input component and wrap with `Controller`.

**Reply by Amstrong** [reply]
Thank you @bluebill1049
